### ISSUE TITLE
Deprecate Hackage package

### ISSUE THREAD

**Issue by infinisil** [NixOS/nixfmt#161]
Nixfmt is currently [on Hackage](https://hackage.haskell.org/package/nixfmt), but I don't think that's necessary:

- While `pkgs.nixfmt` is [currently being populated by Hackage](https://github.com/NixOS/nixpkgs/blob/9b5ed8cb1ed4d0e4411b20ac0f67f33eaae01e97/pkgs/top-level/all-packages.nix#L40327), there's no reason this would be necessary to build a Haskell package in Nix. Indeed the new `pkgs.nixfmt-rfc-style` [works without Hackage](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ni/nixfmt-rfc-style/package.nix)
- Nixfmt doesn't really have an API surface that would be useful enough on its own. I consider it an end-user application. If some useful and reusable API surface emerges from the codebase, we can still consider packaging _that_ on Hackage.

So I'd propose deprecating the Hackage package and not updating it anymore. This way `nixfmt` on Hackage will always be the legacy Serokell version.

This will remove a bunch of maintenance burden from the formatter team.

Ping @Sereja313 @Lucus16 @yorickvP

**Comment by Sereja313** [comment]
I agree

**Comment by dasJ** [comment]
Somebody from the [maintainers team](https://hackage.haskell.org/package/nixfmt/maintainers/) probably has to do this

**Comment by infinisil** [comment]
Yeah I could do that, should I do it already or should we wait until we have a first initial release? It might also be a good idea to update the Hackage page with a bit of information about the deprecation.

**Comment by dasJ** [comment]
We should also move the `nixfmt-classic` package in nixpkgs first to make sure it doesn't point to a non-existing hackage package.

**Comment by jfly** [comment]
We looked at this in today's meeting. We'd like to add a note to the existing hackage package (which will require releasing a 0.6.1). Then we'll mark it deprecated with no replacement.

**Comment by infinisil** [comment]
This is now done: https://hackage.haskell.org/package/nixfmt
